File: C++/1316번 그룹 단어 체커.cpp
#include <iostream>
#include <string>
#include <vector>
using namespace std;

int main() {
    int N, count = 0;
    cin >> N;

    while (N--) {
        string s;
        cin >> s;

        vector<bool> seen(26, false); // 알파벳 사용 여부
        bool is_group = true;

        for (int i = 0; i < s.size(); i++) {
            if (i > 0 && s[i] == s[i-1]) continue;

            if (seen[s[i] - 'a']) {
                is_group = false;
                break;
            }

            seen[s[i] - 'a'] = true;
        }

        if (is_group) count++;
    }

    cout << count << endl;
    return 0;
}

File: Python/1316번 그룹 단어 체커.py
def main():
    count = 0
    N = int(input())
    for _ in range(N):
        s = input()
        group = True
        seen = [False] * 26

        for t in range(len(s)):
            if t > 0 and s[t] == s[t-1]:
                continue
            elif seen[ord(s[t]) - ord('a')]:
                group = False
                break

            seen[ord(s[t]) - ord('a')] = True

        if group:
            count += 1

    print(count)

if __name__ == "__main__":
    main()
